Services/Storage.cs

```diff
@@ -118,6 +118,21 @@ private void AppendBingMapKey(JToken theme)
 
         public async Task<Logo> SetLogoAsync(Logo model)
         {
+            //Do not overwrite existing name or image with null
+            if(model.Name == null || model.Image == null)
+            {
+                Logo current = await this.GetLogoAsync();
+                if(!current.IsDefault)
+                {
+                    model.Name = model.Name ?? current.Name;
+                    if (model.Image == null)
+                    {
+                        model.Image = current.Image;
```

Comment:
is ```current.Image``` not null?

Reply:
It doesn't make a difference since it will set a previously null value to null, but I can add the check for readability